src/sonic-yang-mgmt/setup.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""The setup script."""
+
+from setuptools import setup, find_packages
+from setuptools.command.build_py import build_py
+from os import system, environ
+from sys import exit
+import pytest
+
+# find path of pkgs from os environment vars
+prefix = '../../'; debs = environ["IMAGE_DISTRO_DEBS_PATH"]
+wheels = environ["PYTHON_WHEELS_PATH"]
+wheels_path = '{}/{}'.format(prefix, wheels)
+deps_path = '{}/{}'.format(prefix, debs)
+# dependencies
+libyang = '{}/{}'.format(deps_path, environ["LIBYANG"])
+libyangCpp = '{}/{}'.format(deps_path, environ["LIBYANG_CPP"])
+libyangPy2 = '{}/{}'.format(deps_path, environ["LIBYANG_PY2"])
+libyangPy3 = '{}/{}'.format(deps_path, environ["LIBYANG_PY3"])
+sonicYangModels = '{}/{}'.format(wheels_path, environ["SONIC_YANG_MODELS_PY3"])
+
+# important reuirements parameters
+build_requirements = [libyang, libyangCpp, libyangPy2, libyangPy3, sonicYangModels,]
+
+setup_requirements = ['pytest-runner']
+
+test_requirements = ['pytest>=3']
+
+# read me
+with open('README.rst') as readme_file:
+    readme = readme_file.read()
+
+# class for prerequisites to build this package
+class pkgBuild(build_py):
+    """Custom Build PLY"""
+
+    def run (self):
+        #  install libyang and sonic_yang_models
+        for req in build_requirements:
+            if '.deb' in req:
+                pkg_install_cmd = "sudo dpkg -i {}".format(req)
+                if (system(pkg_install_cmd)):
+                    print("{} installation failed".format(req))
+                    exit(1)
+                else:
+                    print("{} installed".format(req))
+            elif '.whl' in req:
+                pkg_install_cmd = "pip3 install {}".format(req)
+                if (system(pkg_install_cmd)):
+                    print("{} installation failed".format(req))
+                    exit(1)
+                else:
+                    print("{} installed".format(req))
+
```

Comment:
This runs at build time, so it will take PKGs list in 
```build_requirements = [libyang, libyangCpp, libyangPy2, libyangPy3, sonicYangModels,]
```
So we can have a problem only when setup.py is updated specially build_requirements field. I can fail the build if   build_requirements field in setup.py has any other extension. Develop may need to fix it, when they add any other PKG intentionally in build_requirements.

Reply:
Not explicitly complaining as of now, it may be needed only if setup.py is updated in future.